aws-athena/views/sale-vw_ias_salesval_upload.sql

```diff
@@ -0,0 +1,16 @@
+WITH ias_sales AS (
+    SELECT 
+        salekey,
+        NULLIF(REPLACE(instruno, 'D', ''), '') AS instruno_clean
+    FROM {{'iasworld', 'sales'}}
+    WHERE cur = 'Y'
+    AND deactivat IS NULL
+)
+
+SELECT 
+    ias_sales.salekey,
+    sf.sv_is_outlier,
+    sf.sv_outlier_type,
+    sf.run_id
+FROM ias_sales 
+INNER JOIN {{ ref('sale.flag') }} sf ON ias_sales.instruno_clean = sf.meta_sale_document_num;
```

Comment:
The [`build-and-test-dbt` workflow](https://github.com/ccao-data/data-architecture/actions/runs/6384372564/job/17326897801?pr=147) is complaining that the `sale.flag` model does not exist:

```
Compilation Error
Model 'model.athena.sale.vw_ias_salesval_upload' (models/sale/sale.vw_ias_salesval_upload.sql) depends
  on a node named 'sale.flag' which was not found
```

My guess is that `sale.flag` is generated by a Glue job and not by any transformations stored in dbt, is that right? If so, we'll need to treat it as a source (i.e. a table created outside the DAG) instead of a ref (i.e. a table created inside the DAG).

To make this a source, we first need to adjust the call here:

```suggestion
INNER JOIN {{ source('sale', 'flag') }} sf ON ias_sales.instruno_clean = sf.meta_sale_document_num;
```

Then, we'll need to add a `source` configuration to the `models/sale/schema.yml` file below, which I will sketch out in a separate comment.